clang/lib/Sema/SemaBounds.cpp

```diff
@@ -2680,19 +2739,17 @@ namespace {
       case Expr::UnaryOperatorClass: {
         UnaryOperator *UO = cast<UnaryOperator>(E);
         if (UO->getOpcode() == UnaryOperatorKind::UO_Deref)
-          return RValueBounds(UO->getSubExpr(), CSS);
-        else {
-          llvm_unreachable("unexpected lvalue unary operator");
+          return RValueBounds(UO->getSubExpr(), CSS, SideEffects::Disabled);
+        else
```

Comment:
Since CheckUnaryOperator can be called on any kind of unary operator visited by TraverseStmt, it is no longer guaranteed that any calls to LValueBounds made by CheckUnaryOperator will only be on deref operator lvalues. Adding in this unreachable call causes the Clang Sema/complex-imag.c test to fail.